demos/minimal/actors/sample_generator.py

```diff
@@ -15,54 +15,52 @@ class Generator(Actor):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.data = None
         self.name = "Generator"
-        self.frame_num = 0
+        self.frame_num = 0  # Initialize frame counter
+        # Generate xs (x-coordinates)
+        self.xs = np.linspace(-10, 10, 100)
 
     def __str__(self):
-        return f"Name: {self.name}, Data: {self.data}"
+        return f"Name: {self.name}"
 
     def setup(self):
-        """Generates an array that serves as an initial source of data.
-
-        Initial array is a 100 row, 5 column numpy matrix that contains
-        integers from 1-99, inclusive.
-        """
-
+        """Initial setup for Generator."""
         logger.info("Beginning setup for Generator")
-        self.data = np.asmatrix(np.random.randint(100, size=(100, 5)))
         logger.info("Completed setup for Generator")
 
     def stop(self):
-        """Save current randint vector to a file."""
-
+        """Actions to perform on stopping."""
```

Comment:
I think we can maintain the same functionality from before of saving the the data to a file before exiting. 

Just change the code comment to something like "save current wave vector to file" or something similar